Feature/6

StanislavDargel opened the pull request:
#feature6

1. New dependencies added in project. (Spring framework);
2. Were created new package <configuration> and new class ApplicationConfiguration.java;
3. New CloseProgram.java was created, "Exit" method moved from ConsoleUI.java into CloseProgram.java;
4. CreateProduct.java action method was splitted into new classes:
-> ProductCategoryInput.java;
-> ProductDescriptionInput.java;
-> ProductDiscountInput.java;
-> ProductNameInput.java;
-> ProductPriceInput.java;
new Interface ProductParameters.java was added;
5. Minor fixes: Test final variables changed into ordinary variables;
6. Method toString were added into ProductCategory.java, CreateProduct.java, EditParameters.java, FindProductById.java and RemoveProduct.java;
7. Dependencies injections changed by Spring framework into ShoppingListApplication.java;
8. void printProductInfo(...) method changed into String printProductInfo(...) method into ValidationService.java;

StanislavDargel committed:
#AssertionError

StanislavDargel committed:
#feature5
1. Fixed some minor mistakes.
2. In <validation> package exception messages been changed.
3. Project covered with 69% tests.

StanislavDargel committed:
#feature5
1. Fixed some minor mistakes.
2. In <validation> package exception messages been changed.
3. Project covered with 69% tests.

StanislavDargel committed:
#feature5
1. Fixed BeanMapperTest.java, ProductRepositoryTest.java, ValidationServiceTest.java;

StanislavDargel committed:
#feature6
1. New dependencies added in project. (Spring framework);
2. Were created new package <configuration> and new class ApplicationConfiguration.java;
3. New CloseProgram.java was created, "Exit" method moved from ConsoleUI.java into CloseProgram.java;
4. CreateProduct.java action method was splitted into new classes:
-> ProductCategoryInput.java;
-> ProductDescriptionInput.java;
-> ProductDiscountInput.java;
-> ProductNameInput.java;
-> ProductPriceInput.java;
new Interface ProductParameters.java was added;
5. Minor fixes: Test final variables changed into ordinary variables;
6. Method toString were added into ProductCategory.java, CreateProduct.java, EditParameters.java, FindProductById.java and RemoveProduct.java;
7. Dependencies injections changed by Spring framework into ShoppingListApplication.java;
8. void printProductInfo(...) method changed into String printProductInfo(...) method into ValidationService.java;

javagururt commented on src/main/java/com/javaguru/shoppinglist/service/validation/ProductDescriptionValidationRule.java:
```diff
@@ -1,14 +1,16 @@
     public void validate(ProductDTO productDTO) {
         productNotNull(productDTO);
         if (productDTO.getDescription() != null) {
             if (productDTO.getDescription().isEmpty()) {
```
можно использовать isBlank, а так же перенести в 1 строку и использовать &&

javagururt commented on src/main/java/com/javaguru/shoppinglist/service/validation/ProductUniqueValidationRule.java:
```diff
@@ -2,7 +2,9 @@
 
+@Component
 public class ProductUniqueValidationRule implements ProductValidationRule {
     private ProductRepository repository;
```
можно указать final

javagururt commented on src/main/java/com/javaguru/shoppinglist/service/ValidationService.java:
```diff
@@ -6,10 +6,15 @@
+import static java.lang.String.format;
+
+@Service
 public class ValidationService {
```
Это не ValidationService, это скорее ProductService

javagururt commented on src/main/java/com/javaguru/shoppinglist/service/ValidationService.java:
```diff
@@ -63,15 +68,16 @@ private ProductDTO dataNormalizer(ProductDTO productDTO) {
 
-    public void printProductInfo(ProductDTO productDTO) {
-        System.out.printf("\nName: %s\nID: %d\nPrice: %.2f\nCategory: %s%n",
+    public String printProductInfo(ProductDTO productDTO) {
```
подобные методы можно перенести в ProductUtils

javagururt commented on src/test/java/com/javaguru/shoppinglist/mapper/BeanMapperTest.java:
```diff
@@ -1,67 +1,66 @@
 
+@RunWith(MockitoJUnitRunner.class)
 public class BeanMapperTest {
+    private ProductDTO dto = new ProductDTO();
```
это по идее можно удалить. Все равно используется только в методах

StanislavDargel committed:
#feature6
1. New dependencies added in project. (Spring framework);
2. Were created new package <configuration> and new class ApplicationConfiguration.java;
3. New CloseProgram.java was created, "Exit" method moved from ConsoleUI.java into CloseProgram.java;
4. CreateProduct.java action method was splitted into new classes:
-> ProductCategoryInput.java;
-> ProductDescriptionInput.java;
-> ProductDiscountInput.java;
-> ProductNameInput.java;
-> ProductPriceInput.java;
new Interface ProductParameters.java was added;
5. Minor fixes: Test final variables changed into ordinary variables;
6. Method toString were added into ProductCategory.java, CreateProduct.java, EditParameters.java, FindProductById.java and RemoveProduct.java;
7. Dependencies injections changed by Spring framework into ShoppingListApplication.java;
8. void printProductInfo(...) method changed into String printProductInfo(...) method into ValidationService.java;
9. ValidationService.java renamed as ProductService.java;
10. Method printProductInfo(...) moved into ProductInfo.java, <productutils>, from ProductService.java;

StanislavDargel committed:
#feature6
1. New dependencies added in project. (Spring framework);
2. Were created new package <configuration> and new class ApplicationConfiguration.java;
3. New CloseProgram.java was created, "Exit" method moved from ConsoleUI.java into CloseProgram.java;
4. CreateProduct.java action method was splitted into new classes:
-> ProductCategoryInput.java;
-> ProductDescriptionInput.java;
-> ProductDiscountInput.java;
-> ProductNameInput.java;
-> ProductPriceInput.java;
new Interface ProductParameters.java was added;
5. Minor fixes: Test final variables changed into ordinary variables;
6. Method toString were added into ProductCategory.java, CreateProduct.java, EditParameters.java, FindProductById.java and RemoveProduct.java;
7. Dependencies injections changed by Spring framework into ShoppingListApplication.java;
8. void printProductInfo(...) method changed into String printProductInfo(...) method into ValidationService.java;
9. ValidationService.java renamed as ProductService.java;
10. Method printProductInfo(...) moved into ProductInfo.java, <productutils>, from ProductService.java;
11. Method dataNormalizer(...) renamed as normalize and moved into ProductDataNormalizer.java <productutils>, from ProductService.java;
12. Minor fixes in <action> implementations, ConsoleUI.java and ProductInfoTest.java;
13. ProductDataNormalizer.java tested successfully;
